modules/hostap/src/supp_api.c

```diff
@@ -398,6 +398,11 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 		}
 	}
 
+	if (params->bssid[0] != 0) {
```

Comment:
`....00` this is still a valid BSSID, no?

Reply:
It is. But here, we are checking this against NULL character (ASCII value = 0).

Reply:
Yes, so, a BSSID like `00:11:22:33:44:00` would be considered NULL and hence not valid. We should check for all zeroes  to handle the case where user hasn't set or not. And if non-zero, then do basic validation and only then accept.